tests/test_commands_task.py

```diff
@@ -0,0 +1,196 @@
+from appcli.orchestrators import DockerComposeOrchestrator
+
+# ------------------------------------------------------------------------------
+# CONSTANTS
+# ------------------------------------------------------------------------------
+
+APP_NAME = "TEST_APP"
+
+# directory containing this script
+BASE_DIR = Path(__file__).parent
+
+STACK_CONFIGURATION_FILE = Path(BASE_DIR, "resources/test_stack_settings.yml")
+
+DOCKER_COMPOSE_YML = Path(
+    BASE_DIR, "resources/templates/baseline/docker-compose.tasks.yml"
+)
+
+DOCKER_COMPOSE_SERVICES = list(YAML().load(open(DOCKER_COMPOSE_YML))["services"].keys())
+
+# ------------------------------------------------------------------------------
+# TASKS
+# ------------------------------------------------------------------------------
+
+
+class Test_TaskCommands:
+    def test_task_run_headless_arg_long(self, test_env):
+
+        TIME_START = time()
+        result = test_env.invoke_task_command(["run", "--detach", "sleep-1"])
+        TIME_END = time()
+
+        assert TIME_END - TIME_START < 0.1
+        assert result.exit_code == 0
+
+    def test_task_run_headless_arg_short(self, test_env):
+
+        TIME_START = time()
+        result = test_env.invoke_task_command(["run", "-d", "sleep-1"])
+        TIME_END = time()
+
+        assert TIME_END - TIME_START < 0.1
+        assert result.exit_code == 0
+
+    def test_task_run_not_headless(self, test_env):
+
+        TIME_START = time()
+        result = test_env.invoke_task_command(["run", "sleep-1"])
+        TIME_END = time()
+
+        assert TIME_END - TIME_START > 0.1
+        assert result.exit_code == 0
+
+    def test_task_run_invalid_flag(self, test_env):
+
+        result = test_env.invoke_task_command(["run", "-x", "sleep-1"])
+
+        # assert "Error: No such option: -x" in result.output
+        assert result.exit_code == 2
+
+
+# ------------------------------------------------------------------------------
+# Fixtures
+# ------------------------------------------------------------------------------
+
+
+@pytest.fixture(scope="session")
+def test_env(tmp_path_factory):
+    return Environment(tmp_path_factory)
+
+
+@pytest.fixture(autouse=True)
+def patch_subprocess(monkeypatch):
+    def patched_subprocess_run(docker_compose_command, capture_output=True):
+        # TODO: We should take advantage of the printed command to perform test validation
+        logger.info(f"PYTEST_PATCHED_DOCKER_COMPOSE_COMMAND=[{docker_compose_command}]")
+        if not any([x in docker_compose_command for x in ["--detach", "-d", "-x"]]):
+            # patch for running in standard mode.
+            sleep(0.1)
+            return subprocess.CompletedProcess(returncode=0, args=None)
+        elif all([x in docker_compose_command for x in ["-d"]]):
+            # patch for run headless/detached
+            return subprocess.CompletedProcess(returncode=0, args=None)
+        elif all([x in docker_compose_command for x in ["-x"]]):
```

Comment:
Why only this specific argument?

Given my comment above, it may not be worth including this case anyway, since this is likely expected behaviour from the `click` library